ts-client/src/amm/index.ts

```diff
@@ -176,6 +183,193 @@ export default class AmmImpl implements AmmImplementation {
+        invariant(vaultB, `Vault ${poolState.tokenBMint.toBase58()} not found`);
+
+        poolInfoMap.set(poolState.lpMint.toBase58(), {
+          pool,
+          poolState,
+          vaultA,
+          vaultB,
+        });
+        return [
+          { pubkey: apyPda, type: AccountType.APY },
+          { pubkey: vaultA.vaultState.tokenVault, type: AccountType.VAULT_A_RESERVE },
+          { pubkey: vaultB.vaultState.tokenVault, type: AccountType.VAULT_B_RESERVE },
+          { pubkey: vaultA.vaultState.lpMint, type: AccountType.VAULT_A_LP },
+          { pubkey: vaultB.vaultState.lpMint, type: AccountType.VAULT_B_LP },
+          { pubkey: poolState.aVaultLp, type: AccountType.POOL_VAULT_A_LP },
+          { pubkey: poolState.bVaultLp, type: AccountType.POOL_VAULT_B_LP },
+          { pubkey: poolState.lpMint, type: AccountType.POOL_LP_MINT },
+        ];
+      }),
+    );
+
+    const flatAccountsToFetch = accountsToFetch.flat();
+    const accountsBufferMap = await getAccountsBuffer(connection, [
+      ...flatAccountsToFetch,
+      { pubkey: SYSVAR_CLOCK_PUBKEY, type: AccountType.SYSVAR_CLOCK },
+    ]);
+    const accountsInfoMap = deserializeAccountsBuffer(accountsBufferMap);
+
+    const ammImpls = await Promise.all(
+      accountsToFetch.map(async (accounts) => {
+        const [apyPda, tokenAVault, tokenBVault, vaultALp, vaultBLp, poolVaultA, poolVaultB, poolLpMint] = accounts; // must follow order
+        const apy = accountsInfoMap.get(apyPda.pubkey.toBase58()) as ApyState;
+        const currentTime = accountsInfoMap.get(SYSVAR_CLOCK_PUBKEY.toBase58()) as BN;
+        const poolVaultALp = accountsInfoMap.get(poolVaultA.pubkey.toBase58()) as BN;
+        const poolVaultBLp = accountsInfoMap.get(poolVaultB.pubkey.toBase58()) as BN;
+        const vaultALpSupply = accountsInfoMap.get(vaultALp.pubkey.toBase58()) as BN;
+        const vaultBLpSupply = accountsInfoMap.get(vaultBLp.pubkey.toBase58()) as BN;
+        const vaultAReserve = accountsInfoMap.get(tokenAVault.pubkey.toBase58()) as BN;
+        const vaultBReserve = accountsInfoMap.get(tokenBVault.pubkey.toBase58()) as BN;
+        const poolLpSupply = accountsInfoMap.get(poolLpMint.pubkey.toBase58()) as BN;
+
+        invariant(
+          !!apy &&
+            !!currentTime &&
+            !!vaultALpSupply &&
+            !!vaultBLpSupply &&
+            !!vaultAReserve &&
+            !!vaultBReserve &&
+            !!poolVaultALp &&
+            !!poolVaultBLp &&
+            !!poolLpSupply,
+          'Account Info not found',
+        );
+
+        const accountsInfo = {
+          apy,
+          currentTime,
+          poolVaultALp,
+          poolVaultBLp,
+          vaultALpSupply,
+          vaultBLpSupply,
+          vaultAReserve,
+          vaultBReserve,
+          poolLpSupply,
+        };
+
+        const depegAccounts = await getDepegAccounts(ammProgram.provider.connection);
```

Comment:
we should all this outside instead of here, this will call depeg accounts on each pool, inefficient